### DISCUSSION TITLE
Compare files which differ by their extension...

### DISCUSSION THREAD

**Discussion by Izhido** [WinMerge/winmerge#1497] [Q&A]
I do realize this might sound a bit odd, but, please, bear with me.

A project I'm currently working on, based on C++, and therefore containing .cpp files, was forked from another project, containing virtually all the same files, BUT with a .c extension. 

During translation, I wanted to cause as minimal damage as possible to the original project - the original code works 100% OK, I only wanted to be able to take advantage of std::vector & std::unordered_map and other niceties of C++ in some specific areas in it. Thus, most of the C code looks identical to the new C++ code - save for the areas where I use C++-specific stuff.

Recently, I had a need to do a comparison between the original project and the new one, to try and scan differences for a bug I'm attempting to fix. I haven't found yet a way, though, to tell WinMerge "hey, the files to the left are .c, and the files to the right are .cpp, please compare them ignoring their extension".

Is there a way to do this in the latest version of WinMerge? Having that functionality would help me a lot with my current work.

**Comment by sdottaka** [comment]
Unfortunately, WinMerge does not have the ability to ignore differences in file extensions.

How about creating symbolic links from *.cpp files to *.c at the command prompt and comparing them as follows?

~~~
mkdir c:\tmp\src
for %i in (d:\dev\winmerge\src\*.cpp) do mklink c:\tmp\src\%~ni.c %i
~~~
